sentry.gradle

```diff
@@ -192,9 +192,15 @@ static extractBundleTaskArguments(cmdArgs){
     // In this function here, we only grep through the first `metro` step, which only generates an intermediate sourcemap,
     // which is wrong. We need the final one. Luckily, we can just generate the path from the `bundleOutput`, since
     // the paths seem to be well defined.
-    if (bundleOutput != null) {
+    // if sourcemapOutput is not null, it means react-native passed --sourcemap-output and we are good to go
```

Comment:
hm, doesn’t it *always* pass that flag?

At least in the version that I developed against, there were two cases:

1. without hermes:
   * metro -> output.map
2. with hermes:
   * metro -> intermediate1
   * hermes -> intermediate2
   * merging the two -> output.map

We always want to get the `output.map`. How? It seems super fragile to grep and manipulate paths of the `metro` command line. Is there no other way to get that output map?

Reply:
no, it's not always, that's why we check its nullability.
I could not find any other way than `--sourcemap-output` and if it's not there, it's a guessing game, so I check dirs existence.

Reply:
But in its current state, your patch is incorrect, because in case 2 above, it does pass an explicit sourcemap, which in that case would be `intermediate1`, which is the wrong one.
I would recommend to try different alternatives, check that the file exists, and maybe just fall back to the one you detected via the args.

Reply:
which version of RN are you testing with? I have a fresh RN setup and Sentry and never got `intermediate1` coming from `--sourcemap-output`

Reply:
https://github.com/facebook/react-native/blob/25ea35632e18aa41bc753effaf3ad36c1fc84897/react.gradle#L158
I was looking at the `react-native` gradle file all the time. I don’t claim to completely understand it though :-D

Reply:
@Swatinem yep, me too, even the history of it, but again, which RN version were you working on?